src/core/insights/datalayer.ts

```diff
@@ -0,0 +1,15 @@
+declare global {
+  interface Window {
+    dataLayer: unknown[];
+  }
+}
+
+export const track = (event: string, properties?: Record<string, unknown>) => {
+  const dataLayer = window.dataLayer || [];
```

Comment:
Maybe unnecessary, but since we dabble with static building it wouldn't hurt to handle the situation where `window` is not available, i.e. checking `if (typeof window === 'undefined')`.

Reply:
Added a check for this, thanks for pointing it out!